src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantValidator.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Localization;
+using OrchardCore.Data;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Mvc.ModelBinding;
+using OrchardCore.Tenants.ViewModels;
+
+namespace OrchardCore.Tenants.Services
+{
+    public class TenantValidator
```

Comment:
I think an interface-implementation pair would be better.

Reply:
I was thinking about that first, then I decided to remove it because the service can be mocked easily

Reply:
It doesn't seem to me that it can. It's a concrete implementation with non-virtual public methods.

Reply:
One more thing if using `TenantValidator` without interface will allow us to add `IsDefaultShell` as `internal` method. I know you can woraround it by casting. I don't see value to introduce an interface here

Let me push my changes, then we can figure out if it's better to add an interface or not

Reply:
You can make `IsDefaultShell()` part of the interface. I don't see why we'd need a concrete implementation for that.

Reply:
It doesn't make sense, while it's a validator service, we should think from APIs perspective

Reply:
You can have it on `ShellSettings` instead then.

Reply:
Make sense